src/macros-by-example.adoc

```diff
@@ -993,19 +983,18 @@ parameter produces a stream of structs.
 The binary encoding of macro-shaped parameters are similarly tagless, eliding any opcodes
 mentioning `point` and just writing its arguments with minimal delimiting.
 
-Macro types can be grouped and/or combined with cardinality modifiers, following the same rules as
-tagless types.  Note that grouped macro types require callers to use two layers of delimiting
-containers: and outer list for the group, and an inner S-expression for
-each macro instance:
+Macro types can be combined with cardinality modifiers, with invocations using groups
+as needed.  Note that grouping macro types require callers to use two layers of delimiting
+containers:
 
 [{nrm}]
 ----
 (*macro* scatterplot
-  [(points [point] *+*), (x_label *string*), (y_label *string*)]
+  [(points point *+*), (x_label *string*), (y_label *string*)]
   { points: [points], x_label: x_label, y_label: y_label })
 ----
 ----
-(:scatterplot [(3 17), (395 23), (15 48), (2023 5)] "hour" "widgets")
+(:scatterplot (: (3 17) (395 23) (15 48) (2023 5)) "hour" "widgets")
```

Comment:
1. What would you think about having some syntax in the text encoding that is specific for macro-type arguments... something that looks like a macro invocation but has some symbol that indicates that the name is redundant in that position? (Strawman suggestion: `(:*` ... `)`.) It might help improve readability in cases where there are also s-exps in the macro arguments.
2. Suppose `point` can accept multiple coordinates, then one could have something like this?
```
(:scatterplot (: (3 (:17 18 19)) (395 23) (15 48) (2023 5)) "hour" "widgets")
```

Reply:
I think this is something else that a more visually distinctive expression group would help to address.

> What would you think about having some syntax in the text encoding that is specific for macro-type arguments... something that looks like a macro invocation but has some symbol that indicates that the name is redundant in that position?

Something I really like about the expression group is that its contents are always--for any encoding, including macro-shaped--tied to the corresponding parameter's declared encoding. When an exactly-one parameter has a macro-shaped encoding, the text representation of that encoding is an s-expression containing the arguments to the declared macro. When there's a macro-shaped expression group, it's an expression group that contains zero or more instances of the same sexp-of-args syntax.

I think making macro-shaped expression groups into yet another construct would be added complexity for limited benefit.

Reply:
I think adding syntax for macro-encoded arguments defeats the purpose, which is to improve brevity.  I think

```
(:foo (: (:* 1 2) (:* 3 4) (:* 5 6)))
```
is significantly less human-friendly than:
```
(:foo (: (1 2) (3 4) (5 6)))
```

The concept also
* intersects with the ungrouped case: does one write `(:foo (1 2))` or `(:foo (:* 1 2))`?
* requires additional tokenization rules following the `(:*`